internal/commands/app/create.go

```diff
@@ -228,25 +231,34 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 		}
 	}
 
-	if dsCluster.Name != "" {
-		local.AddDataSource(appLocal.AppData, map[string]interface{}{
-			"name": dsCluster.Name,
-			"type": dsCluster.Type,
-			"config": map[string]interface{}{
-				"clusterName":         dsCluster.Config.ClusterName,
-				"readPreference":      dsCluster.Config.ReadPreference,
-				"wireProtocolEnabled": dsCluster.Config.WireProtocolEnabled,
-			},
-		})
+	clusterNames := make([]string, 0, len(dsClusters))
+	for _, dsCluster := range dsClusters {
+		if dsCluster.Name != "" {
```

Comment:
is this condition possible still?  seems like not

if it is, i would only suggest inverting the `if` clause here and using `continue`, so we can un-indent the `local.AddDataSource` block below

```go
if dsCluster.Name == "" { continue }
local.addDataSource(...)
```